### ISSUE TITLE
Memory Leak with Safari

### ISSUE THREAD

**Issue by bradley-varol** [gruhn/vue-qrcode-reader#82]
Hi,

Thanks for a great package, but I have encountered a memory leak with Safari on iOS 11.4.1.
The `qrcode-reader` component is being used like so:

```js
<qrcode-reader :paused="paused" @decode="onDecode" @init="onInit"></qrcode-reader>
```

When using the Timeline in Safari's Web Inspector, I'm noticing that while the component is active, Javascript memory usage rises up to several gigabytes before crashing Safari with the "A problem occurred with this webpage so it was reloaded." alert. 
Javascript "Animation Frame" events containing `t > drawImage` and `getImageData` are firing several times a second, with some of these events taking longer than 30ms to execute.
In 30 seconds, 127 "Message Event Dispatched" events containing `flushCallbacks` are also firing, taking an average of 193ms to execute.

I hope that this is sufficient information. Cheers 👍

**Comment by gruhn** [comment]
Hey, thanks for reporting this.

Did you encounter this issue on the demo page. Or in your own project?

Unfourtunately, I have no iOS device so we have to investigate this together.

**Comment by gruhn** [comment]
ok, cool. 

Make sure to check out both [Decode continuously](https://gruhn.github.io/vue-qrcode-reader/demos/DecodeAll.html) and [Disable tracking](https://gruhn.github.io/vue-qrcode-reader/demos/NoTracking.html) which has visual tracking disabled. It processes frames [only twice per second](https://github.com/gruhn/vue-qrcode-reader/blob/0d231329c82ab5ea398a7e3b46d8327e45e289bd/src/components/QrcodeStream.vue#L82-L88). If this demo reveals memory leaks too, you should at least take longer to crash.

**Comment by bradley-varol** [comment]
> ok, cool.
> 
> Make sure to check out both [Decode continuously](https://gruhn.github.io/vue-qrcode-reader/demos/DecodeAll.html) and [Disable tracking](https://gruhn.github.io/vue-qrcode-reader/demos/NoTracking.html) which has visual tracking disabled. It processes frames [only twice per second](https://github.com/gruhn/vue-qrcode-reader/blob/0d231329c82ab5ea398a7e3b46d8327e45e289bd/src/components/QrcodeStream.vue#L82-L88). If this demo reveals memory leaks too, you should at least take longer to crash.

:( The memory is rising on the docs too... After 2 minutes, I'm at 3GB.
I'll test the two things you mentioned. Thanks!

**Comment by bradley-varol** [comment]
I'd also like to note here for you that before disabling tracking, a qr code would sometimes be scanned after unpausing. It would only happen 1 in maybe 10 scans, and the decoded string would be the same as the scan so assume it's accidentally scanning twice. Disabling tracking has fixed this, probably due to the increased frames between scans.

**Comment by gruhn** [comment]
I fixed a memory leak in [v1.3.3](https://github.com/gruhn/vue-qrcode-reader/releases/tag/v1.3.3). I don't really think it's related to this issue but check it out anyway.

**Comment by gruhn** [comment]
Closing this due to inactivity. Please reopen if the issue persists.
